contracts/XVSVault/XVSVault.sol

```diff
@@ -187,6 +190,29 @@ contract XVSVault is XVSVaultStorage, ECDSA {
         emit Deposit(msg.sender, _rewardToken, _pid, _amount);
     }
 
+    /**
+     * @notice Claim rewards for pool
+     * @param _account The account for which to claim rewards
+     * @param _rewardToken The Reward Token Address
+     * @param _pid The Pool Index
+     */
+    function claim(address _account, address _rewardToken, uint256 _pid) external nonReentrant {
+        _ensureValidPool(_rewardToken, _pid);
+        PoolInfo storage pool = poolInfos[_rewardToken][_pid];
+        UserInfo storage user = userInfos[_rewardToken][_pid][_account];
+        _updatePool(_rewardToken, _pid);
+        if (user.amount > 0) {
+            uint256 pending = user.amount.sub(user.pendingWithdrawals).mul(pool.accRewardPerShare).div(1e12).sub(
+                user.rewardDebt
+            );
+
+            user.rewardDebt = user.amount.sub(user.pendingWithdrawals).mul(pool.accRewardPerShare).div(1e12);
+
+            IXVSStore(xvsStore).safeRewardTransfer(_rewardToken, _account, pending);
+            emit Claim(_account, _rewardToken, _pid, pending);
```

Comment:
Nit: maybe we could move this logic into a separate function and reuse it in `deposit` and `requestWithdrawal`?

I it would require substantial changes, we could add this `Claim` event to `deposit`/`requestWithdrawal` functions, because claiming happens there automatically.

Reply:
Actually we first calculate the reward and transfer to the user then we execute the deposit/withdraw logic and then we calculate the `user.rewardDebt`. The `rewardDebt` is dependent on the final balance of user after deposit/withdrawal. So we will need two separate functions. 

I have added `Claim` event to `deposit`/`requestWithdrawal`